### DISCUSSION TITLE
v2.0.0 Technical Preview 5

### DISCUSSION THREAD

**Discussion by the1812** [the1812/Bilibili-Evolved#2075] [分享 / Share]
- 元数据中添加了 `@connect localhost`, 解决 Tampermonkey BETA 中不能从本地安装组件的问题.
- 修复 `LifeCycleEventTypes.ComponentsLoaded` 在组件运行完成前就触发的问题.
- 动态 API 支持自定义内容过滤器. (`动态过滤器`基于此 API 实现了对顶栏动态的过滤)
- 搜索栏的历史同步 b 站的搜索历史 (同 v1), 之前产生的历史数据可以通过运行以下代码来删除.
```js
delete bilibiliEvolved.settingsApis.getComponentSettings('launchBar').options.searchHistory
```
- 支持设置 `文件下载模式`.

**插件新增:**

- 下载视频 - IDM导出
- 搜索栏 - 搜索推荐

**组件迁移:**

所有常用组件已迁移完成, 剩余未迁移的还有:
- 工作量过大 (Coming S∞n)
  - 清爽首页
  - 极简首页
- 隐藏功能 (有一定缺陷所以未公开)
  - 网址AV号转换
  - 评论楼层显示
- 废弃功能 (被 b 站官方实现)
  - 默认视频画质
  - 解除音量上限

另外一部分组件功能还未更新到 v1.12.12 的进度:
- 下载视频
  - 不支持 ffmpeg 相关导出
  - 不支持课程 / 番剧 / 电影
- 自定义顶栏
  - 分区未更新
  - 搜索栏在搜索页没有同步搜索词
  - 历史面板中的直播没有状态显示
  - 收藏面板还不能记住上次选择的收藏夹
  - 收藏面板还不能显示已失效视频
  - 视频动态还不能显示发布时间
  - `排行`还没有子菜单
  - UI 样式未统一

<hr /><em>This discussion was created from the release <a href='https://github.com/the1812/Bilibili-Evolved/releases/tag/v2.0.0-tp5'>v2.0.0 Technical Preview 5</a>.</em>

**Comment by kethoven** [comment]
自定义顶栏 - 夜间模式开关 好像用不了。

感觉新版本功能组件一个一个手动添加，太麻烦了。能不能把所有功能集成在一起。手动卸载自己不需要的功能啊。

**Reply by the1812** [reply]
> 自定义顶栏 - 夜间模式开关 好像用不了。

你是不是装到组件里去了, 我还没写判断, 这个要装在插件里的

> 感觉新版本功能组件一个一个手动添加，太麻烦了。

TP6 会有合集包, 可以批量安装多个功能

**Reply by kethoven** [reply]
> > 自定义顶栏 - 夜间模式开关 好像用不了。
> 
> 你是不是装到组件里去了, 我还没写判断, 这个要装在插件里的
> 
> > 感觉新版本功能组件一个一个手动添加，太麻烦了。
> 
> TP6 会有合集包, 可以批量安装多个功能

谢谢，我都装到组件去了。新版本是还不太会用。

**Reply by Geeyun-JY3** [reply]
> 感觉新版本功能组件一个一个手动添加，太麻烦了。

别急啊，这不还在 TP 嘛，没正式发布麻烦一点很正常：https://github.com/the1812/Bilibili-Evolved/tree/v2#设置
> 如果你曾经使用过 v1 版, 可以利用 `v1 设置迁移` 组件将旧设置导入到 v2 中, 该工具将自动把里面开启的设置对应的组件下载并安装. (当然, 现在也还没做出来)

**Reply by kethoven** [reply]
> > 感觉新版本功能组件一个一个手动添加，太麻烦了。
> 
> 别急啊，这不还在 TP 嘛，没正式发布麻烦一点很正常：[https://github.com/the1812/Bilibili-Evolved/tree/v2#设置](https://github.com/the1812/Bilibili-Evolved/tree/v2#%E8%AE%BE%E7%BD%AE)
> 
> > 如果你曾经使用过 v1 版, 可以利用 `v1 设置迁移` 组件将旧设置导入到 v2 中, 该工具将自动把里面开启的设置对应的组件下载并安装. (当然, 现在也还没做出来)

好的！慢慢完善，慢慢适应吧

**Comment by hhh12395** [comment]
批量添加组件/插件？
关于界面添加跳转到项目页面 ? (方便看到更新日记)
